gorp.go

```diff
@@ -66,22 +66,28 @@ type TypeConverter interface {
 // See the DbMap function docs for each of the functions below for more
 // information.
 type SqlExecutor interface {
-	WithContext(ctx context.Context) SqlExecutor
-	Get(i interface{}, keys ...interface{}) (interface{}, error)
-	Insert(list ...interface{}) error
-	Update(list ...interface{}) (int64, error)
-	Delete(list ...interface{}) (int64, error)
-	Exec(query string, args ...interface{}) (sql.Result, error)
-	Select(i interface{}, query string, args ...interface{}) ([]interface{}, error)
-	SelectInt(query string, args ...interface{}) (int64, error)
-	SelectNullInt(query string, args ...interface{}) (sql.NullInt64, error)
-	SelectFloat(query string, args ...interface{}) (float64, error)
-	SelectNullFloat(query string, args ...interface{}) (sql.NullFloat64, error)
-	SelectStr(query string, args ...interface{}) (string, error)
-	SelectNullStr(query string, args ...interface{}) (sql.NullString, error)
-	SelectOne(holder interface{}, query string, args ...interface{}) error
-	Query(query string, args ...interface{}) (*sql.Rows, error)
-	QueryRow(query string, args ...interface{}) *sql.Row
+	Get(ctx context.Context, i interface{}, keys ...interface{}) (interface{}, error)
+	Insert(ctx context.Context, list ...interface{}) error
+	Update(ctx context.Context, list ...interface{}) (int64, error)
+	Delete(ctx context.Context, list ...interface{}) (int64, error)
+	Select(ctx context.Context, i interface{}, query string, args ...interface{}) ([]interface{}, error)
+	SelectInt(ctx context.Context, query string, args ...interface{}) (int64, error)
+	SelectNullInt(ctx context.Context, query string, args ...interface{}) (sql.NullInt64, error)
+	SelectFloat(ctx context.Context, query string, args ...interface{}) (float64, error)
+	SelectNullFloat(ctx context.Context, query string, args ...interface{}) (sql.NullFloat64, error)
+	SelectStr(ctx context.Context, query string, args ...interface{}) (string, error)
+	SelectNullStr(ctx context.Context, query string, args ...interface{}) (sql.NullString, error)
+	SelectOne(ctx context.Context, holder interface{}, query string, args ...interface{}) error
+
+	// These method signatures are shared with *sql.DB.
+	// Stylistically, `Context` in the name is redundant. It is the future, everything takes a context.
+	// But since these three functions delegate to functions of the same name on *sql.DB, it would be
+	// a little confusing if we had, e.g. `Exec` (taking a context), which delegates to
+	// `sql.DB.ExecContext` (taking a context) as opposed to `sql.DB.Exec` (taking no context).
+	// So we don't bother with `Context` in the name for Get, Insert, etc., but we do for these.
+	ExecContext(ctx context.Context, query string, args ...interface{}) (sql.Result, error)
+	QueryContext(ctx context.Context, query string, args ...interface{}) (*sql.Rows, error)
+	QueryRowContext(ctx context.Context, query string, args ...interface{}) *sql.Row
```

Comment:
(Hyper-personal preference time) I find the confusion of "why do these three methods have a different name from everything else" to be much greater than the confusion of "why does this method Exec delegate to a method called ExecContext". I think this is largely because I don't expect gorp's methods to be direct passthroughs to sql.DB: none of the other methods in this interface are!

For a minute I thought the much stronger argument was going to be "but if the method names are different, then the `executor` interface below won't work!". But it turns out that's not the case: the only types that are ever used to satisfy the `executor` interface are borp.DbMap and borp.Transaction. So in fact I think this sways things the other way: the fact that sql.DB *can* satisfy the `executor` interface feels dangerous!

Reply:
> I don't expect gorp's methods to be direct passthroughs to sql.DB

https://pkg.go.dev/github.com/go-gorp/gorp#DbMap.Exec

> ... This is equivalent to running: Exec() using database/sql

https://pkg.go.dev/github.com/go-gorp/gorp#DbMap.Prepare

> ... This is equivalent to running: Prepare() using database/sql

For Query and QueryRow, neither gorp nor borp currently have any documentation. We should add some, but I think the reason they are currently undocumented is that by following the pattern, one can assume they pass through to sql.Query and sql.QueryRow.

Reply:
Fair enough. I think it would be totally appropriate for us to simply update the borp documentation to say "this is equivalent to running: ExecContext() using database/sql".